webapi/src/main/scala/org/knora/webapi/routing/v2/OntologiesRouteV2.scala

```diff
@@ -919,107 +710,69 @@ final case class OntologiesRouteV2(
       get { requestContext =>
-        val propertyIri = propertyIriStr.toSmartIri
-        stringFormatter.checkExternalOntologyName(propertyIri)
-
-        if (!propertyIri.getOntologySchema.contains(ApiV2Complex)) {
-          throw BadRequestException(s"Invalid class IRI for request: $propertyIriStr")
-        }
-
-        val requestMessageFuture: Future[CanDeletePropertyRequestV2] = for {
-          requestingUser <- getUserADM(requestContext)
-        } yield CanDeletePropertyRequestV2(
-          propertyIri = propertyIri,
-          requestingUser = requestingUser
-        )
-
-        RouteUtilV2.runRdfRouteF(requestMessageFuture, requestContext)
+        val requestMessageTask = for {
+          propertyIri <- RouteUtilZ
+                           .toSmartIri(propertyIriStr, s"Invalid property IRI: $propertyIriStr")
+                           .flatMap(RouteUtilZ.ensureExternalOntologyName)
+                           .flatMap(RouteUtilZ.ensureApiV2ComplexSchema)
+          requestingUser <- Authenticator.getUserADM(requestContext)
+        } yield CanDeletePropertyRequestV2(propertyIri, requestingUser)
+        RouteUtilV2.runRdfRouteZ(requestMessageTask, requestContext)
       }
     }
 
   private def deleteProperty(): Route =
     path(ontologiesBasePath / "properties" / Segments) { externalPropertyIris: List[IRI] =>
       delete { requestContext =>
-        val propertyIriStr = externalPropertyIris match {
-          case List(str) => str
-          case _         => throw BadRequestException(s"Only one property can be deleted at a time")
-        }
-
-        val propertyIri = propertyIriStr.toSmartIri
-        stringFormatter.checkExternalOntologyName(propertyIri)
-
-        if (!propertyIri.getOntologySchema.contains(ApiV2Complex)) {
-          throw BadRequestException(s"Invalid property IRI for request: $propertyIri")
-        }
-
-        val lastModificationDateStr = requestContext.request.uri
-          .query()
-          .toMap
-          .getOrElse(LAST_MODIFICATION_DATE, throw BadRequestException(s"Missing parameter: $LAST_MODIFICATION_DATE"))
-        val lastModificationDate = ValuesValidator
-          .xsdDateTimeStampToInstant(lastModificationDateStr)
-          .getOrElse(throw BadRequestException(s"Invalid timestamp: $lastModificationDateStr"))
-
-        val requestMessageFuture: Future[DeletePropertyRequestV2] = for {
-          requestingUser <- getUserADM(requestContext)
+        val requestMessageTask = for {
+          propertyIri <-
+            ZIO
+              .succeed(externalPropertyIris)
+              .filterOrFail(_.size == 1)(BadRequestException(s"Only one property can be deleted at a time"))
+              .map(_.head)
+              .flatMap(iri => RouteUtilZ.toSmartIri(iri, s"Invalid property IRI: $iri"))
+              .flatMap(RouteUtilZ.ensureExternalOntologyName)
+              .flatMap(RouteUtilZ.ensureApiV2ComplexSchema)
```

Comment:
multiple flatmaps within a for-comprehension make me suspicious... could this be re-written to a more direct syntax along those lines?

```suggestion
    iris <- ZIO.succeed(externalPropertyIris)
    iri  <- iris.filterOrFail(_.size == 1)(BadRequestException(s"Only one property can be deleted at a time")).map(_.head)
    smartIri    <- RouteUtilZ.toSmartIri(iri, s"Invalid property IRI: $iri")
    _           <- RouteUtilZ.ensureExternalOntologyName(smartIri)
    _           <- RouteUtilZ.ensureApiV2ComplexSchema(smartIri)
```

Reply:
(same goes for the following similar instances... could this maybe even be factored out?)

Reply:
IMHO this reads more nicely because it is entirely clear for which iri the `ensure` functions are used.

In your example you have to read and understand that `smarIri` is passed into the function:
```
   smartIri    <- RouteUtilZ.toSmartIri(iri, s"Invalid property IRI: $iri")
    _           <- RouteUtilZ.ensureExternalOntologyName(smartIri)
    _           <- RouteUtilZ.ensureApiV2ComplexSchema(smartIri)
```

whereas in the original code it is immediately obvious:
```
   smartIri    <- RouteUtilZ.toSmartIri(iri, s"Invalid property IRI: $iri")
                         .flatMap(RouteUtilZ.ensureExternalOntologyName)
                         .flatMap(RouteUtilZ.ensureApiV2ComplexSchema)
```
 An improvement could be to extract the whole thing into a a method. I have considered that but decided against it for now. This is because it is hard to name this method also because there are so many different combinations of check in the other routes and I liked to be able to directly see which `ensure` checks are done on an iri passed in from the route.